libs/filamat/src/MaterialBuilder.cpp

```diff
@@ -967,10 +961,30 @@ Package MaterialBuilder::build(JobSystem& jobSystem) noexcept {
     return package;
 }
 
-const std::string MaterialBuilder::peek(filament::backend::ShaderType type,
+bool MaterialBuilder::hasCustomVaryings() const noexcept {
+    for (const auto& variable : mVariables) {
+        if (!variable.empty()) {
+            return true;
+        }
+    }
+    return false;
+}
+
+bool MaterialBuilder::needsStandardDepthProgram() const noexcept {
```

Comment:
By "standard" here you mean vertex+fragment we use for the color pass, so not the "depth" program right?

Reply:
No actually, I meant the standard vertex shader + the depth fragment shader.

At no time do we use the standard (ie color) fragment shader with the DEPTH variant.

I think this is correct, there is nothing in the standard fragment shader that we need when we're only doing the depth.

Reply:
All we sometimes need is alpha testing (we need to discard based on alpha). As long as we have that test when needed we're good.

Reply:
Ah yes, `depth_main.fs` does that indeed:
```
#if defined(BLEND_MODE_MASKED)
    if (alpha < getMaskThreshold()) {
        discard;
    }
#endif
```